python/pyiceberg/catalog/hive.py

```diff
@@ -197,17 +234,41 @@ def __init__(self, name: str, properties: Properties, uri: str):
         super().__init__(name, properties)
         self._client = _HiveClient(uri)
 
-    def _convert_hive_into_iceberg(self, table: HiveTable) -> Table:
-        # Requires reading the manifest, will implement this in another PR
-        # Check the table type
-        return Table(identifier=(table.dbName, table.tableName))
+    def _convert_hive_into_iceberg(self, table: HiveTable, metadata_location: Optional[str] = None) -> Table:
+        properties: Dict[str, str] = table.parameters
+        if TABLE_TYPE not in properties or properties[TABLE_TYPE] != ICEBERG:
+            raise NoSuchTableError(f"Table is not an Iceberg table: {table.dbName}.{table.tableName}")
```

Comment:
Should we add "missing table_type" to this?

Reply:
Sure thing. I've also split the exception into two parts; when the property `table_type` is missing, and when it is not equal to `iceberg`.